src/network/address.rs

```diff
@@ -86,17 +86,17 @@ impl Encodable for Address {
 
 impl Decodable for Address {
     #[inline]
-    fn consensus_decode<D: io::Read>(mut d: D) -> Result<Self, encode::Error> {
+    fn consensus_decode<R: io::Read>(r: &mut R) -> Result<Self, encode::Error> {
         Ok(Address {
-            services: Decodable::consensus_decode(&mut d)?,
-            address: read_be_address(&mut d)?,
-            port: u16::swap_bytes(Decodable::consensus_decode(d)?)
+            services: Decodable::consensus_decode(r)?,
+            address: read_be_address(r)?,
+            port: u16::swap_bytes(Decodable::consensus_decode(r)?)
         })
     }
 }
 
 /// Read a big-endian address from reader.
-fn read_be_address<R: io::Read>(mut r: R) -> Result<[u16; 8], encode::Error> {
+fn read_be_address<R: io::Read>(mut r: &mut R) -> Result<[u16; 8], encode::Error> {
```

Comment:
```suggestion
fn read_be_address<R: io::Read>(r: &mut R) -> Result<[u16; 8], encode::Error> {
```

Requires removing the `& mut` a few lines below in the code also.